tests/parametric/test_partial_flushing.py

```diff
@@ -79,14 +78,17 @@ def do_partial_flush_test(self, test_agent, test_library):
     """
     with test_library:
         with test_library.start_span(name="root") as parent_span:
-            with test_library.start_span(name="child1", parent_id=parent_span.span_id):
+            with test_library.start_span(name="child1", parent_id=parent_span.span_id) as child1:
                 pass
-            partial_trace = test_agent.wait_for_num_traces(1, clear=True, wait_loops=30)
-            child_span = find_span_in_traces(partial_trace, Span(name="child1"))
+            partial_traces = test_agent.wait_for_num_traces(1, clear=True, wait_loops=30)
+            partial_trace = find_trace(partial_traces, parent_span.trace_id)
             assert len(partial_trace) == 1
+            child_span = find_span(partial_trace, child1.span_id)
             assert child_span["name"] == "child1"
+
     traces = test_agent.wait_for_num_traces(1, clear=True)
```

Comment:
I don't understand this test 😅 . Why is this test calling `test_agent.wait_for_num_traces(1, ...)` twice? It is because we expect two trace chunks? Why not use `test_agent.wait_for_num_traces(2, ...)` instead?

And what is `partial_trace` vs `full_trace`? Does `test_agent` return the entire trace (the two trace chunks combined) in the second call to `test_agent.wait_for_num_traces()`? That seems weird. How can we test that each span was sent in it own trace chunk?

Reply:
Ideally, a test for partial flushing with `DD_TRACE_PARTIAL_FLUSH_MIN_SPANS=1` and two spans should:
- assert that the tracer sent exactly two trace chunks
- assert that each trace chunk has exactly one span
- assert that the span from one chunk is the parent and the span from the other chunk is the child

The order of the chunks or the spans is not defined, so the test shouldn't which trace chunk contains the parent span or the child span.

Reply:
Yup, the test agent combines all trace chunks into one list and returns it: https://github.com/DataDog/dd-apm-test-agent/blob/master/ddapm_test_agent/agent.py#L425. The easiest way to test partial flushing is to call `wait_for_num_traces` twice 😭 (or we can check whether the spans are marked as top level but that's less reliable).

Reply:
Oh, I see 😬  Thanks for clarifying!

Reply:
We may need to change that someday to allow testing "chunk tags" like `_dd.p.tid`, but not here, not now 😅